agent/job_runner.go

```diff
@@ -392,6 +394,28 @@ func (r *JobRunner) normalizeVerificationBehavior(behavior string) (string, erro
 	}
 }
 
+func filterEnv(environmentVariables map[string]string, allowedVariables []string) []string {
+	// Filter out if the user has unlisted environment variables
+	// It adds environment variable to denied list and removes if it finds matching regex
+	var deniedVariables []string
```

Comment:
I would like to see the consistency, than mixing. Either we call things:
allowedEnv / deniedEnv OR allowedVariables / deniedvariables